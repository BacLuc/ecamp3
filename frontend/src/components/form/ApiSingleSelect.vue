<!--
Displays a field as text or as an input field, depending on the editing prop.
You can two-way bind to the value using v-model.
-->

<template>
  <api-wrapper
    v-bind="$props">
    <template slot-scope="wrapper">
      <v-select
        v-model="wrapper.localValue"
        v-bind="$attrs"
        :label="label"
        :error-messages="wrapper.errorMessage"
        @input="wrapper.on.input"
        @blur="wrapper.on.touch">
        <status-icon slot="append" :status="wrapper.status" />
      </v-select>
    </template>
  </api-wrapper>
</template>

<script>
import { apiPropsMixin } from '@/mixins/apiPropsMixin'
import ApiWrapper from './ApiWrapper'
import StatusIcon from './StatusIcon'

export default {
  name: 'ApiSingleSelect',
  components: { ApiWrapper, StatusIcon },
  mixins: [apiPropsMixin],
  props: {
    // disable delay in default for select component
    autoSaveDelay: { type: Number, default: 0, required: false }
  },
  data () {
    return {
    }
  }
}
</script>

<style scoped>
</style>
