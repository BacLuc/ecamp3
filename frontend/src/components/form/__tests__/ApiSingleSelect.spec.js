// Libraries
import Vue from 'vue'
import Vuetify from 'vuetify'

import { mount, createLocalVue } from '@vue/test-utils'
import ApiSingleSelect from '../ApiSingleSelect.vue'

// const localVue = createLocalVue()
// localVue.use(Vuetify)

Vue.use(Vuetify)

describe('ApiTextField.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  test('renders correctly', () => {
    const props = {
      value: 'Test Value',

      /* field name and URI for saving back to API */
      fieldname: 'test-field',
      uri: 'test-field/123',

      /* display label */
      label: 'Test Field',

      /* overrideDirty=true will reset the input if 'value' changes, even if the input is dirty. Use with caution. */
      overrideDirty: false,

      /* enable/disable edit mode */
      editing: true,

      /* enable/disable auto save */
      autoSave: true,

      /* Validation criteria */
      required: true
    }

    const wrapper = mount(ApiSingleSelect, {
      // localVue,
      vuetify,
      propsData: props
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
