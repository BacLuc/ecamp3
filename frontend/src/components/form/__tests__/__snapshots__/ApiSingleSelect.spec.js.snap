// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiTextField.vue renders correctly 1`] = `
<span>
  <!---->
   
  <span>
    <form
      class="v-form mb-2"
      inline=""
      novalidate="novalidate"
    >
      <div
        class="v-input theme--light v-text-field v-select"
      >
        <div
          class="v-input__control"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="list-8"
            class="v-input__slot"
            role="button"
          >
            <div
              class="v-select__slot"
            >
              <label
                class="v-label theme--light"
                for="input-8"
                style="left: 0px; position: absolute;"
              >
                Test Field
              </label>
              <div
                class="v-select__selections"
              >
                <input
                  aria-readonly="true"
                  autocomplete="off"
                  id="input-8"
                  readonly="readonly"
                  type="text"
                />
              </div>
              <div
                class="v-input__append-inner"
              >
                <span>
                  <!---->
                   
                  <!---->
                </span>
              </div>
              <input
                type="hidden"
                value="Test Value"
              />
            </div>
            <div
              class="v-menu"
            >
              <div
                class="v-menu__content theme--light "
                style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
              />
            </div>
          </div>
          <div
            class="v-text-field__details"
          >
            <div
              class="v-messages theme--light"
            >
              <div
                class="v-messages__wrapper"
              />
            </div>
          </div>
        </div>
      </div>
       
      <!---->
       
      <!---->
    </form>
  </span>
</span>
`;
